server/storage/rethinkdb.go

```diff
@@ -15,6 +14,13 @@ import (
 	"gopkg.in/dancannon/gorethink.v3"
 )
 
+// RethinkDB has eventual consistency. This represents a 60 second blackout
+// period of the most recent changes in the changefeed which will not be
+// returned while the eventual consistency works itself out.
+// It's a var not a const so that the tests can turn it down to zero rather
+// than have to include a sleep.
+var blackoutTime = 60
```

Comment:
Why a blackout instead of a quorum read, since we use `gorethink.TableOpts{ReadMode: "majority"}` for `GetCurrent` calls?

Reply:
This gives some additional protection in the case of a partition. Also, even with majority reads and writes, we could still end up with a little inconsistency on the most recent few records due to clock skews etc...

Reply:
> This gives some additional protection in the case of a partition.

Is in the goal is to make change feeds always readable so long as even one node in an n-node cluster is up?

> we could still end up with a little inconsistency on the most recent few records due to clock skews etc

Could you explain how that would happen?

Reply:
As I understand it, bearing in mind much of RethinkDB is a mystery to me, during a partition (I think during leader election too) we may have records being written to DBs that aren't synchronized while the partition resolves. After resolution those records end up interleaved. That means you'll get an inconsistent feed during vs after the event.

As far as clock skew, I'm using the Go server time but I don't think the rethinkdb instances are guaranteed to have perfectly synchronized clocks either. Server A inserts a record at time 12345, Server B inserts a record at 12344 according to the servers clock, but in the real world the even happens after Server A's insert.

We don't have to worry about clock skew in the MySQL version because we use the database's time and there is only 1 replica.

Reply:
I'm also somewhat hazy on what RethinkDB does, especially since they don't use consensus for the data, just table metadata.  But going by their consistency guarantees here:  https://rethinkdb.com/docs/consistency/, we may want to specify majority reads for change feeds anyway, even with the blackout for better consistency?

According to https://aphyr.com/posts/329-jepsen-rethinkdb-2-1-5, which is a bit old but I don't think the model is outdated, only the primary replica can make changes (guaranteeing serializability of data), and the primary is selected via raft, so I'm not sure if for a single table results can be interleaved as a result of a partition (since only the primary can cause writes).

I can see what you mean about the clock skew though, although if I'm reading this correctly, we order by `CreatedAt`, which is determined by the notary server's clock, as opposed to the rethink server's clock?  Do we perform any checks on the notary server that would guarantee that the clock skew does not exceed 60 seconds?

Reply:
No, we're relying on NTP or some custom mechanism to keep all the notary servers reasonably in sync.